plugins/scaffolder-backend/src/scaffolder/tasks/StorageTaskBroker.ts

```diff
@@ -271,4 +271,33 @@ export class StorageTaskBroker implements TaskBroker {
     this.deferredDispatch.resolve();
     this.deferredDispatch = defer();
   }
+
+  async abort(taskId: string) {
```

Comment:
Bit confused here, there's nothing that actually aborts the task right? The task execution itself is just gonna continue with an abort event in the middle?

As a user I would expect that any logs that are shown is what actually happened, and nothing more. It feels like an implementation of this would need to signal to the workflow runner that the user requested to the task to be aborted, and as soon as the runner sees that it will halt execution and complete the task. It could be neat to have that handled through an AbortController too, but a minimal implementation would probably be that the runner checks the state before every step.

Reply:
You are right. Initially I thought it would be hard to implement with current architecture and intentionally left it out. 
But with a slight tweak, I found a way to revive it back.

Tested by running the task on instance 1 and aborting the task on instance 2.

Reply:
🙄

Reply:
Tried this out locally now too with two instances. The frontend works looks fine and it seems like stuff is working, but in the backend it keeps executing both the current and any remaining actions. This means the user would think that a repo never got published for example, but in reality it did.

Reply:
To be able to abort the current action, each action has to be abortable. 
`AbortController` provides a callback to abort, but we need to implement it in each action to actually abort. 
I didn't go do that as it will blow the PR even more. 
We can document that current action is not going to be aborted unless this functionality is provided by the action. 
As it also applies to custom actions.

Regarding follow up actions, I wonder how do you test? 
As for me it works well.

I do run 2 instances by using 2 different start scripts

<img width="1328" alt="Screenshot 2022-11-28 at 16 29 14" src="https://user-images.githubusercontent.com/2265094/204316713-a13d5c85-f569-4e5e-9b4a-694cd95b47ac.png">

Where I point in each to different ports in backend and frontend. 

<img width="1505" alt="Screenshot 2022-11-28 at 16 30 08" src="https://user-images.githubusercontent.com/2265094/204316950-c4477700-163c-45fe-ad1d-086c385a14bd.png">

As an example template I took this one:

<img width="415" alt="Screenshot 2022-11-28 at 16 30 44" src="https://user-images.githubusercontent.com/2265094/204317121-e921db17-d0b0-48c2-87c5-9d16a617924b.png">

Modified a bit `publish:github` step to have enough time to abort it from another instance.

<img width="1087" alt="Screenshot 2022-11-28 at 16 31 45" src="https://user-images.githubusercontent.com/2265094/204317350-2fd08d45-beed-4c6e-87d9-8a8f2b067a82.png">

Added `console.log` in the steps loop to see how many steps are executed

Reply:
Yep I did more or less the same test, except I had a custom wait action that logged during execution, and I ran the action multiple times in multiple steps. When triggering the run in the first backend and aborting via the other, the UI updates straight away and says it's cancelled. The wait action keeps logging, which is expected since I didn't use any signal, but it also kept executing all the remaining wait actions too.

I do think what we need here is more of a handshake mechanism, where we don't write to the action state straight away that it's cancelled, but rather mark it for cancellation and then have the workflow runner be able to detect that and carry out the cancellation.